src/Utilities/cachingoptimizer.jl

```diff
@@ -270,15 +270,19 @@ end
 # setting the same type of set or function.
 function replace_constraint_function_or_set!(m::CachingOptimizer, attr, cindex, replacement)
     if m.state == AttachedOptimizer
-        if m.mode == Automatic
-            try
-                MOI.set!(m.optimizer, attr, m.model_to_optimizer_map[cindex], replacement)
-            catch err
-                if err isa MOI.CannotSetAttribute
-                    resetoptimizer!(m)
-                else
-                    rethrow(err)
+        if m.mode == Automatic            
+            if MOI.supports(m.optimizer, attr, typeof(cindex))
```

Comment:
I would rather just add `|| (err isa MOI.UnsupportedAttribute)` in the `catch` call to be consistent with
https://github.com/JuliaOpt/MathOptInterface.jl/blob/45095a5accb1ec6a9f8a37b7cf628df91e9cc9af/src/Utilities/cachingoptimizer.jl#L254

Reply:
But here we have an additional lookup in the dictionary that can be avoided since `cindex` is of the same type as `m.model_to_optimizer_map[cindex]`.

Reply:
If `supports` return `false` then `resetoptimizer!` (long) will be called and this happens only once every `optimize` call (rare). We don't want a negligible optimization in a rare case, the case we want to make fast is when this is supported. In that case, it's better not to call `MOI.supports` as it might account for a nonnegligible time